src/postings/recorder.rs

```diff
@@ -125,18 +117,22 @@ impl Recorder for DocIdRecorder {
         let (buffer, doc_ids) = buffer_lender.lend_all();
         self.stack.read_to_end(arena, buffer);
         // TODO avoid reading twice.
+        let mut prev_doc = 0;
         if let Some(doc_id_map) = doc_id_map {
-            doc_ids.extend(
-                VInt32Reader::new(&buffer[..])
-                    .map(|old_doc_id| doc_id_map.get_new_doc_id(old_doc_id)),
-            );
+            doc_ids.extend(VInt32Reader::new(&buffer[..]).map(|delta_doc_id| {
```

Comment:
+ 1 in favor of scan of a loop.

Having `prev_doc` outside of the if-statement is not a good idea. Having it in both statement makes the code easier   to (proof)read.

Reply:
switched to scan, but we can't apply the same technique in the other cases, because they contain a mix of delta and non delta encoded values